packages/admin/blocks-admin/src/blocks/helpers/createCompositeBlockSelectField.tsx

```diff
@@ -0,0 +1,35 @@
+import { Field, FieldProps, FinalFormSelect } from "@comet/admin";
+import { MenuItem } from "@mui/material";
+import * as React from "react";
+import { FormProps } from "react-final-form";
+
+import { BlocksFinalForm } from "../../form/BlocksFinalForm";
+import { createCompositeSetting } from "./composeBlocks/createCompositeSetting";
+
+interface Options<T extends string | number> {
```

Comment:
Do we need this extends? Could there be cases where this doesn't work, e.g., enums?

Reply:
Good point with the enums. I wanted to restrict the use to only string and number but this would hinder enums to be used. I'll remove it 👍🏻

Reply:
Ah yeah now I remember. If I don't do that the key prop throws this massive error:
<img width="1314" alt="image" src="https://github.com/vivid-planet/comet/assets/56400587/f0ab0ebf-0957-41a1-a1d9-45301c6309c8">

Reply:
SelectField also has this constraint